metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -307,6 +310,24 @@ final class ReferenceProvider(
       result
     }
   }
+  override def references(
+      buildTargetIdentifier: String,
+      symbol: String,
+  ): Integer = {
+    val visited = scala.collection.mutable.Set.empty[AbsolutePath]
+    index.iterator.count { case (path, entry) =>
+      if (
+        entry.id.getUri == buildTargetIdentifier
+        && entry.bloom.mightContain(symbol)
+      ) {
+        val sourcePath = AbsolutePath(path)
+        if (!visited(sourcePath)) {
+          visited.add(sourcePath)
+          sourcePath.exists
+        } else false
```

Comment:
```suggestion
        if (visited.add(sourcePath)) {
          sourcePath.exists
        } else false
```
This could be simplified to check whether the given element already belongs to the set only once. :slightly_smiling_face:

Reply:
made it shorter: `visited.add(sourcePath) && sourcePath.exists`